### PR TITLE
rebase tcs_workorder_receipt_handler on json-rpc

### PR DESCRIPTION
**Pull request by sammyne**
1. this is the final part following PR #106
2. replace the manually unwrapping requests and wrapping response with json-rpc

Signed-off-by: lixiangmin01 <lixiangmin01@baidu.com>

### CODE REVIEW

File: examples/common/python/connectors/direct/tcs_listener/tcs_workorder_receipt_handler.py [hunk 1]
```diff
@@ -48,299 +64,218 @@ def __workorder_receipt_on_boot(self):
...
+                "Work order receipt already exist in the database. Hence"
+                + "invalid parameter")
+
+        value = self.kv_helper.get("wo-requests", wo_id)
+        if not value:
+            raise JSONRPCDispatchException(
+                WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                "Work order does not exists. Hence invalid parameter")
+
+        # santize params
```
[the remark this answers is not in the document]
Comment (sammyne) [reply examples/common/python/connectors/direct/tcs_listener/tcs_workorder_receipt_handler.py]: done~

**Comment by sammyne** [comment]
> * One minor comment in `tcs_workorder_receipt_handler.py`
> * I assume the deleted `workorder_receipt_handler() functionality was added in a previous PR

`workorder_receipt_handler()` is not added by me~

**Comment by danintel** [comment]
@sammyne  
> workorder_receipt_handler() is not added by me~ @danintel
I know it is not added by you--you are deleting it from file `tcs_workorder_receipt_handler.py `

Why is `workorder_receipt_handler()` no longer needed?

**Comment by sammyne** [comment]
> @sammyne
> 
> > workorder_receipt_handler() is not added by me~ @danintel
> > I know it is not added by you--you are deleting it from file `tcs_workorder_receipt_handler.py `
> 
> Why is `workorder_receipt_handler()` no longer needed?

`workorder_receipt_handler()` is responsible for routing requests to specific handlers, and this job is now placed on the  JSON-RPC Dispatcher in the TCSListener @danintel

**Comment by Ram-srini** [comment]
@sammyne I have added full fledged work order receipt APIs in my PR#162. Can you please review it. I have added request validation in the receipt APIs. If you think PR#162 is good then you can close your PR.

**Comment by sammyne** [comment]
> @sammyne I have added full fledged work order receipt APIs in my PR#162. Can you please review it. I have added request validation in the receipt APIs. If you think PR#162 is good then you can close your PR.

Nice work 😄 @Ram-srini

**Comment by danintel** [comment]
Link to PR #162 Work order receipt APIs implementation for direct model

**Pull request closed without merging**